tests/test_eks/test_eks.py

```diff
@@ -0,0 +1,770 @@
+from __future__ import unicode_literals
+
+from copy import deepcopy
+from datetime import datetime
+
+import boto3
+import mock
+import pytest
+import sure  # noqa
+
+from moto.eks.exceptions import (
+    ResourceNotFoundException,
+    ResourceInUseException,
+    InvalidRequestException,
+    InvalidParameterException,
+)
+from moto.eks.models import (
+    NODEGROUP_EXISTS_MSG,
+    CLUSTER_NOT_READY_MSG,
+    CLUSTER_NOT_FOUND_MSG,
+    CLUSTER_IN_USE_MSG,
+    LAUNCH_TEMPLATE_WITH_DISK_SIZE_MSG,
+    LAUNCH_TEMPLATE_WITH_REMOTE_ACCESS_MSG,
+    NODEGROUP_NOT_FOUND_MSG,
+    CLUSTER_EXISTS_MSG,
+)
+from moto.utilities.utils import random_string
+from test_eks_constants import (
+    BatchCountSize,
+    ClusterAttributes,
+    ClusterInputs,
+    DISK_SIZE,
+    INSTANCE_TYPES,
+    LAUNCH_TEMPLATE,
+    MethodNames,
+    NodegroupAttributes,
+    NodegroupInputs,
+    PageCount,
+    PARTITIONS,
+    RegExTemplates,
+    REMOTE_ACCESS,
+    ResponseAttribute,
+    REGION,
+    SERVICE,
+    TestResults,
+)
+from test_eks_utils import (
+    attributes_to_test,
+    generate_clusters,
+    generate_nodegroups,
+    is_valid_uri,
+    random_names,
+    region_matches_partition,
+)
+
+from moto import mock_eks
+from moto.core import ACCOUNT_ID
+from moto.core.utils import iso_8601_datetime_without_milliseconds
+from moto.eks.responses import DEFAULT_MAX_RESULTS
+
+
+@pytest.fixture(scope="function")
+def ClusterBuilder():
+    class ClusterTestDataFactory:
+        def __init__(self, count, minimal):
+            # Generate 'count' number of random Cluster objects.
+            self.cluster_names = generate_clusters(client, count, minimal)
+
+            # Get the name of the first generated Cluster.
+            first_name = self.cluster_names[0]
+
+            # Collect the output of describe_cluster() for the first Cluster.
+            self.cluster_describe_output = client.describe_cluster(name=first_name)[
+                ResponseAttribute.CLUSTER
+            ]
+
+            # Pick a random Cluster name from the list and a name guaranteed not to be on the list.
+            (self.existing_cluster_name, self.nonexistent_cluster_name) = random_names(
+                self.cluster_names
+            )
+
+            # Generate a list of the Cluster attributes to be tested when validating results.
+            self.attributes_to_test = attributes_to_test(
+                ClusterInputs, self.existing_cluster_name
+            )
+
+    def _execute(count=1, minimal=True):
+        return client, ClusterTestDataFactory(count, minimal)
+
+    mock_eks().start()
+    client = boto3.client(SERVICE)
```

Comment:
Can we specify a region name here? Don't really have a preference on which region, as long as it runs on machines without a default. Same for the other 2 invocations.

`boto3.client(SERVICE, region_name=DEFAULT_REGION)`